cfme/common/provider.py

```diff
@@ -153,47 +152,153 @@ def create(self, cancel=False, validate_credentials=True, check_existing=False,
+        if self.one_of(InfraProvider):
+            main_view = navigate_to(self, 'Edit')
+            # todo: to replace/merge this code with create
+            # update values:
+            # filling main part of dialog
+            if updates:
+                main_view.fill(updates)
+
+            # filling endpoints
+            if endpoints:
+                endpoints = endpoints if isinstance(endpoints, list) else [endpoints, ]
+
+                for endpoint in endpoints:
+                    endp_view = getattr(self.endpoints_form(parent=main_view), endpoint.name)
+                    endp_view.fill(endpoint.view_value_mapping)
+
+                    # filling credentials
+                    cur_endpoint = next(
+                        endp for endp in self.endpoints if endp.name == endpoint.name)
+                    if hasattr(endpoint, 'credentials'):
+                        if not hasattr(cur_endpoint, 'credentials') or \
+                                endpoint.credentials != cur_endpoint.credentials:
+                            if hasattr(endp_view, 'change_password'):
+                                endp_view.change_password.click()
+                            elif hasattr(endp_view, 'change_key'):
+                                endp_view.change_key.click()
+                            else:
+                                NotImplementedError(
+                                    "Such endpoint doesn't have change password/key button")
+
+                            endp_view.fill(endpoint.credentials.view_value_mapping)
+                            if validate_credentials:
+                                endpoint.view.validate.click()
+
+            if cancel:
+                main_view.cancel.click()
+                cancel_text = 'Edit of Infrastructure Provider "{name}" ' \
+                              'was cancelled by the user'.format(name=self.name)
+                flash.assert_message_match(cancel_text)
+            else:
+                main_view.save.click()
+                if endpoints:
+                    self.endpoints = endpoints
+                if updates:
+                    self.name = updates.get('name', self.name)
+
+                if BZ.bugzilla.get_bug(1436341).is_opened and version.current_version() > '5.8':
+                    logger.warning('Skipping flash message verification because of BZ 1436341')
+                    return
+
+                # todo: replace flash when main view is merged
+                flash.assert_message_match(
+                    '{} Provider "{}" was saved'.format(self.string_name, self.name))
+        else:
+            navigate_to(self, 'Edit')
+            fill(self.properties_form, self._form_mapping(**updates))
+            for cred in self.credentials:
+                fill(self.credentials[cred].form, updates.get('credentials', {}).get(cred, None),
+                     validate=validate_credentials)
```

Comment:
This issue happens because of bug in web_ui forms for rhos7-ga. As far as I recall it can't be easily fixed. It will be definitely fixed during conversion to widgetastic what I will do soon.
Additionally, it isn't related to my changes at all.